IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -47,71 +47,121 @@ - (instancetype)init
     return self;
 }
 
-- (void)storeCredential:(MSIDCredentialCacheItem *)credential forKey:(MSIDCacheKey *)key
+- (void)storeItem:(id<MSIDJsonSerializable>)item forKey:(MSIDCacheKey *)key
 {
     dispatch_barrier_async(self.queue, ^{
-        [self.cacheObjects setObject:credential forKey:key];
+        NSString *type = [self getItemTypeFromCacheKey:key];
+        
+        if (type)
+        {
+            NSMutableDictionary *items = nil;
```

Comment:
nit: you can set items here already and avoid extra objectForKey: call

e.g.

NSMutableDictionary *items = [self.cacheObjects objectForKey:type];

if (!items)
{
items = [NSMutableDictionary new];
}

[items setObject:item forKey:key];
[self.cacheObjects setObject:items forKey:type];

Reply:
Agree. Addressed.